File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size();
        vector <int>resultsl(n,1);
        vector <int>resultsr(n,1);

        int left = 1;
        for(int i =0;i<resultsl.size();i++){
            
            resultsl[i] = left;
            left = left*nums[i];

        }
        int right = 1;
        for(int i =n-1;i>=0;i--){
            resultsr[i] = right;
            right = right*nums[i];

        }
        

        for(int i =0;i<n;i++){
            resultsl[i] *= resultsr[i];

        }

        return resultsl;
    }
};

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.py


class Solution:
    def levelOrder(self, root):
        if not root:
            return []
        

        queue = deque()
        result = []
        queue.append(root)

        while queue:
            level = []
            for i in range(len(queue)):
                node = queue.popleft()
                level.append(node.val)

                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)

            result.append(level)
        return result
